source/core/slang-uint-set.h

```diff
@@ -117,7 +141,93 @@ class UIntSet
         /// Returns true if set1 and set2 have a same value set (ie there is an intersection)
     static bool hasIntersection(const UIntSet& set1, const UIntSet& set2);
 
+    struct Iterator
+    {
+        friend class UIntSet;
+    private:
+        const List<Element>* context;
+        Index block = 0;
+        Element processedElement = 0;
+    
+    public:
+        Iterator(const List<Element>* inContext) 
+        {
+            context = inContext;
+        }
+
+        Element operator*()
+        {
+            Element bitUnset = processedElement;
+            processedElement &= processedElement - 1;
```

Comment:
We shouldn't be modifying the state of the iterator at operator*. This will lead to incorrect result if operator* is called in succession.
The only place we can update the iterator is in ++.

Reply:
this is a mistake 👍 
I forgot to check if UIntSet::Iterator worked before I pushed.